src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -172,38 +88,32 @@ public IList<PostEvictionCallbackRegistration> PostEvictionCallbacks
         public long? Size
         {
             get => _size;
-            set
-            {
-                if (value < 0)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} must be non-negative.");
-                }
-
-                _size = value;
-            }
+            set => _size = !value.HasValue || value >= 0 ? value : throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} must be non-negative.");
         }
 
         public object Key { get; private set; }
 
         public object Value
         {
             get => _value;
-            set
-            {
-                _value = value;
-                _valueHasBeenSet = true;
-            }
+            set { _value = value; IsValueSet = true; }
```

Comment:
Why have 2 lines of code on the same line? I think this makes it harder to read.